src/Handler/CommandNameExtractor/NamedCommandExtractor.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace League\Tactician\Handler\CommandNameExtractor;
+
+/**
+ * Extract the name from a NamedCommand
+ */
+class NamedCommandExtractor implements CommandNameExtractor
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getNameForCommand($command)
+    {
+        if ($command instanceof NamedCommand) {
+            return $command->getName();
+        }
+
+        // fallback to class name
+        return get_class($command);
```

Comment:
Also, I am thinking about adding a fallback here anyway. I mean, probably not all commands implement the NamedCommand interface. So this extractor could accept another in the constructor (optionally) and if it is there, the extractor could use that instead of throwing an exception